d2compiler/compile.go

```diff
@@ -420,6 +420,45 @@ func (c *compiler) compileReserved(attrs *d2graph.Attributes, f *d2ir.Field) {
 		attrs.GridColumns = &d2graph.Scalar{}
 		attrs.GridColumns.Value = scalar.ScalarString()
 		attrs.GridColumns.MapKey = f.LastPrimaryKey()
+	case "grid-gap":
+		v, err := strconv.Atoi(scalar.ScalarString())
+		if err != nil {
+			c.errorf(scalar, "non-integer grid-gap %#v: %s", scalar.ScalarString(), err)
+			return
+		}
+		if v < 0 {
+			c.errorf(scalar, "grid-gap must be a non-negative integer: %#v", scalar.ScalarString())
+			return
+		}
+		attrs.GridGap = &d2graph.Scalar{}
+		attrs.GridGap.Value = scalar.ScalarString()
```

Comment:
I wonder if you should just set `HorizontalGap` and `VerticalGap` here, and then `grid-gap` would be just a syntax sugar for `vertical-gap = horizontal-gap = x`.
The idea is that you don't need to make this check again in the future (if `grid-gap` is set)

Reply:
you can't, because then order would matter.
```
grid-gap: 5
vertical-gap: 10
```

would the not mean the same as

```
vertical-gap: 10
grid-gap: 5
```

Reply:
actually i guess vertical-gap can always override, and grid-gap would not write if already written...

Reply:
yeah sgtm @gavin-ts let's do that

Reply:
you can't override grid-gap then

```
grid-gap: 5
grid-gap: 10
```

would remain 5

Reply:
oh also good point. k nvm

Reply:
> you can't, because then order would matter.
> 
> ```
> grid-gap: 5
> vertical-gap: 10
> ```
> 
> would the not mean the same as
> 
> ```
> vertical-gap: 10
> grid-gap: 5
> ```

To some extent this looks like an error, as `vertical-gap` should have stronger preference, for being specific, over `grid-gap`